pkg/util/test/util.go

```diff
@@ -1713,10 +1716,16 @@ func validatePvcControllerPorts(annotations map[string]string, pvcControllerDepl
 					if len(container.Command) > 0 {
 						for _, containerCommand := range container.Command {
 							if strings.Contains(containerCommand, "--secure-port") {
-								if len(pvcSecurePort) == 0 {
-									return nil, true, fmt.Errorf("failed to validate secure-port, secure-port is missing from annotations in the StorageCluster, but is found in the PVC Controler pod %s", pod.Name)
-								} else if pvcSecurePort != strings.Split(containerCommand, "=")[1] {
-									return nil, true, fmt.Errorf("failed to validate secure-port, wrong --secure-port value in the command in PVC Controller pod [%s]: expected: %s, got: %s", pod.Name, pvcSecurePort, strings.Split(containerCommand, "=")[1])
+								if isAKS(cluster) {
+									if strings.Split(containerCommand, "=")[1] != AksPVCControllerSecurePort {
+										return nil, true, fmt.Errorf("failed to validate secure-port, secure-port is missing in the PVC Controler pod %s", pod.Name)
+									}
+								} else {
+									if len(pvcSecurePort) == 0 {
+										return nil, true, fmt.Errorf("failed to validate secure-port, secure-port is missing from annotations in the StorageCluster, but is found in the PVC Controler pod %s", pod.Name)
+									} else if pvcSecurePort != strings.Split(containerCommand, "=")[1] {
+										return nil, true, fmt.Errorf("failed to validate secure-port, wrong --secure-port value in the command in PVC Controller pod [%s]: expected: %s, got: %s", pod.Name, pvcSecurePort, strings.Split(containerCommand, "=")[1])
+									}
```

Comment:
If the `portworx.io/pvc-controller-secure-port` is set, we need to overwrite the port even if it is AKS. So the correct testing order should be like below - 
```
if (pvcSecurePort) == 0 {
  if isAKS(cluster) && if strings.Split(containerCommand, "=")[1] != AksPVCControllerSecurePort {
      return nil, true, fmt.Errorf("failed to validate secure-port, secure-port on the PVC Controler pod %s is invalid", pod.Name)
  }
  return nil, true, fmt.Errorf("failed to validate secure-port, secure-port is missing from annotations in the StorageCluster, but is found in the PVC Controler pod %s", pod.Name) 
} else if pvcSecurePort != strings.Split(containerCommand, "=")[1] {
  ...
}
```